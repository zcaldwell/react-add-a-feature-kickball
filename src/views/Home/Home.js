import React from 'react';
import { Link } from 'react-router-dom';
import kick from '../../assets/kick.png';
import './Home.css';

function Home({ user }) {
  return (
    <div className="home">
      {user ? (
        <>
          <h1>Hello, {user.email}!</h1>
          <img src={kick} alt="leg kicking ball" style={{ marginTop: '10px', width: '512px' }} />
        </>
      ) : (
        <>
          <h2>Welcome to the Kickball League Directory!</h2>
          <p>
            Please <Link to="/sign-in">sign in</Link> to continue.
          </p>
        </>
      )}
    </div>
  );
}

export default Home;
