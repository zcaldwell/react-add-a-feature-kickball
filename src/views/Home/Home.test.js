import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';

it('should render a welcome message if not logged in', () => {
  const { container } = render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  expect(screen.getByText('Welcome to the Kickball League Directory!')).not.toBeEmptyDOMElement();
  expect(container).toMatchSnapshot();
});

it('should render a message with the userâ€™s email if logged in', () => {
  const { container } = render(
    <MemoryRouter>
      <Home user={{ email: 'test@example.com' }} />
    </MemoryRouter>
  );

  expect(screen.getByText('Hello, test@example.com!')).not.toBeEmptyDOMElement();
  expect(container).toMatchSnapshot();
});
