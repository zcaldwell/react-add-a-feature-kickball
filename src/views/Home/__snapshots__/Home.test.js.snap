// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a message with the userâ€™s email if logged in 1`] = `
<div>
  <div
    class="home"
  >
    <h1>
      Hello, 
      test@example.com
      !
    </h1>
    <img
      alt="leg kicking ball"
      src="kick.png"
      style="margin-top: 10px; width: 512px;"
    />
  </div>
</div>
`;

exports[`should render a welcome message if not logged in 1`] = `
<div>
  <div
    class="home"
  >
    <h2>
      Welcome to the Kickball League Directory!
    </h2>
    <p>
      Please 
      <a
        href="/sign-in"
      >
        sign in
      </a>
       to continue.
    </p>
  </div>
</div>
`;
